app/src/org/commcare/android/framework/SaveSessionCommCareActivity.java

```diff
@@ -0,0 +1,36 @@
+package org.commcare.android.framework;
+
+import android.view.Menu;
+import android.view.MenuItem;
+
+import org.commcare.android.session.DevSessionRestorer;
+import org.javarosa.core.services.locale.Localization;
+
+/**
+ * Any CC activity from which we want the user to be able to save the current session via an item
+ * in the options menu should subclass this class
+ *
+ * @author amstone
+ */
+public abstract class SaveSessionCommCareActivity<R> extends SessionAwareCommCareActivity<R> {
+
+    // Choose a high number so that it will be higher than the menu ids of all other menu items
+    // created by subclasses
+    private static final int MENU_SAVE_SESSION = 9999;
```

Comment:
Whoa so this actually doesn't work, it causes an IndexOutOfBounds exception in some instances (unclear why only sometimes, I only just encountered it when I tried adding onPrepareOptionsMenu() and access the item at index 9999). I think I have to make this id Menu.First, and then have all the other classes that subclass this start their menu numbering at Menu.First + 1